posts/setting-up-a-shared-library-and-seed-job-in-jenkins-2.md

```diff
@@ -0,0 +1,189 @@
+
+## Introduction
+In this second part of the series, we will be setting up a [Jenkins Shared library](https://jenkins.io/doc/book/pipeline/shared-libraries/) to execute our Jenkins jobs. As the complexity and number of jobs you maintain grow, the use of Shared Libraries provides the ability to organize and version control your CI and CD processes. 
+
+In order to set up a practical application with our `seedJob`, we will modify `seed.groovy` to build a Pipeline job and a Multibranch Pipeline job. By the end of this series, you will have a foundation set up to onboard projects consistently as well as handle job specific stages during the running of your Jenkins jobs. 
+
+### Source Code
+The source code is available below
+  * Jenkins Shared Library ([`microservice-pipelines`](https://github.com/kcrane3576/microservice-pipelines))
+  * [`poc-micro`](https://github.com/kcrane3576/poc-micro) [JHipster](https://www.jhipster.tech/) microservice
+
+## Prerequisites
+1. Jenkins running in Docker a docker container built from `kcrane121/maven-jenkins:blog`.
+2. `seedJob` set up from Part 1.
+
+## Part 2 Goals
+1. Configure Jenkins to use [`microservice-pipelines`](https://github.com/kcrane3576/microservice-pipelines)
+as our Shared Library for executing jobs. 
+2. Configure `seed.groovy` to create a Pipeline Job and Multibranch Pipeline Job per service.
+3. Create `jenkinsJob.groovy` to be used by our services as the entry point into our Shared Library.
+4. Set up a [`Jenkinsfile`](https://jenkins.io/doc/book/pipeline/jenkinsfile/) in `poc-micro` to use `microservice-pipelines` Shared Library.
+
+## Goal 1
+### Configure default Shared Library setup for Jenkins
+Since we will be using a Shared library, Jenkins needs to be set up to use our Shared Library.
+
+   1. Navigate to `Dashboard` > select `Manage Jenkins` > select `Configure System` > scroll down to `Global Pipeline Libraries` > select `Add`
+   2. Enter `microservice-pipelines` in the `Name` field
+   3. Ented `master` in `Default Version`
+      * This tells jenkins which branch of our Shared Library we plan to use by default.
+   4. Under `Source Code Management`, select `Git` 
+   5. In the `Project Repository` field, enter `https://github.com/kcrane3576/microservice-pipelines` > select `Save`.
+   ![jenkins shared library configuration](hhttps://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-config-2.png)
+
+## Goal 2
+We are going to modify `seed.groovy` to build a Pipeline job and Multibranch Pipeline job for all services we oboard. 
+
+#### Update `seedJob` to use a `part2` branch we will create in `microservice-pipelines`
+1. Navitate to `Dashboard` > select `seedJob` > select `Configure`.
```

Comment:
navigate

Reply:
The latest version has this and other type-o's corrected